internal/repository/evmreader.go

```diff
@@ -239,3 +245,130 @@ func (pg *Database) GetLastProcessedBlock(
 
 	return block, nil
 }
+
+func (pg *Database) GetPreviousSubmittedClaimsEpochs(
+	ctx context.Context,
+	app Address,
+	block uint64,
+) ([]*Epoch, error) {
+	query := `
+	SELECT
+		id,
+		application_address,
+		index,
+		first_block,
+		last_block,
+		claim_hash,
+		transaction_hash,
+		status
+	FROM
+		epoch
+	WHERE
+		application_address=@appAddress AND status=@status AND last_block <= @block
+	ORDER BY
+		index ASC`
+
+	args := pgx.NamedArgs{
+		"appAddress": app,
+		"status":     EpochStatusClaimSubmitted,
+		"block":      block,
+	}
+
+	rows, err := pg.db.Query(ctx, query, args)
+	if err != nil {
+		return nil, fmt.Errorf("GetProcessedEpochs failed: %w", err)
+	}
+
+	var (
+		id, index, firstBlock, lastBlock uint64
+		appAddress                       Address
+		claimHash, transactionHash       *Hash
+		status                           string
+		results                          []*Epoch
+	)
+
+	scans := []any{
+		&id, &appAddress, &index, &firstBlock, &lastBlock, &claimHash, &transactionHash, &status,
+	}
+	_, err = pgx.ForEachRow(rows, scans, func() error {
+		epoch := &Epoch{
+			Id:              id,
+			Index:           index,
+			AppAddress:      appAddress,
+			FirstBlock:      firstBlock,
+			LastBlock:       lastBlock,
+			ClaimHash:       claimHash,
+			TransactionHash: transactionHash,
+			Status:          EpochStatus(status),
+		}
+		results = append(results, epoch)
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("GetProcessedEpochs failed: %w", err)
+	}
+	return results, nil
+}
+
+func (pg *Database) UpdateEpochsClaimStatusTransaction(
+	ctx context.Context,
+	app Address,
+	claims []*Epoch,
```

Comment:
I'd suggest adding some documentation to the function.

Also:
```suggestion
func (pg *Database) UpdateEpochsStatus(
	ctx context.Context,
	app Address,
	epochs []*Epoch,
```